### PR TITLE
Fixes #1428 Converts command arguments to environment variables and back again (merge to release/1.3)

### LINKED ISSUE #1428: Remove user credentials from Marathon command line
Problem:
Currently Marathon allows to add HTTP user credentials (ID & password) via --http-credentials command line parameter. When the process is running anyone can easily see these credentials in the "ps" command output.

Proposed solutions:
1. Modify --http-credentials parameter once the process is started so it does not display in the "ps" output.
2. Read credentials from a file (preferred). This way more than one user can be defined.
3. Authenticate users against LDAP (even better).

Thanks
Nikolay

### PR DESCRIPTION
**Pull request by brad-peters**
This change allows environment variables to be passing into marathon so they do not show up in ps.

### CODE REVIEW

File: bin/marathon-framework [hunk 1]
```diff
@@ -75,7 +100,6 @@ function load_options_and_log {
       cmd+=( --master "$(cat /etc/mesos/zk)" )
     fi
   fi
-  echo "${cmd[@]}"
```
Comment (brad-peters) [inline bin/marathon-framework]: Some commands/environment variables are secrets we do not want logged

**Comment by jasongilanfarr** [comment]
ok to test

**Comment by iderdik** [comment]
What is the intent here. Environment variables are still visible via `docker inspect`? What could make this more helpful is better support for treating environment variables as secrets and encrypting them.

**Comment by brad-peters** [comment]
The intent was the arguments for the marathon process itself not marathon apps.

The bash script inside of the marathon_runnable.jar converts environment variables to command line arguments for the marathon jar.  So they appear when you run ps.  

Perhaps this doesn't solve both problems in that issue, only the marathon process half.  Maybe there should have been 2 different issues

File: src/main/scala/mesosphere/marathon/Main.scala [hunk 2]
```diff
@@ -29,12 +29,30 @@ class MarathonApp extends App {
     )
   }
 
-  override val conf = new AllConf(args)
+  private val envPrefix = "MARATHON_"
```
Comment (jasongilanfarr) [inline src/main/scala/mesosphere/marathon/Main.scala]: Constant should be "EnvPrefix" as per scala style guidelines.

File: src/main/scala/mesosphere/marathon/Main.scala [hunk 3]
```diff
@@ -29,12 +29,30 @@ class MarathonApp extends App {
...
+  private val envArgs: Array[String] = {
```
Comment (jasongilanfarr) [inline src/main/scala/mesosphere/marathon/Main.scala]: Prefer `Seq[]` please. You should be able to do something like

`
sys.env.withFilter(_._1.startsWith(EnvPrefix)).flatMap { case (key, value) => Seq(s"--${key.replaceAll(EnvPrefix, "").toLowerCase.trim}", if (value.trim.length > 0) value else Nil) }(collection.breakOut)
`

**Comment by brad-peters** [comment]
@jasongilanfarr I have another branch with these changes based off of master, should I update this merge request with that branch?

**Comment by jasongilanfarr** [comment]
@brad-peters Yes please. I'm open to merging into the releases branch as well, but let's first have it in master.

**Pull request merged**